### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'text'

### ISSUE THREAD

**Issue by BananaLoaf** [OrganicIrradiation/scholarly#37]
```bash
Python 3.7.3 (default, Mar 26 2019, 21:43:19) 
[GCC 8.2.1 20181127] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scholarly
>>> print(next(scholarly.search_author('Steven A. Cholewiak')))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.7/site-packages/scholarly/scholarly.py", line 110, in _search_citation_soup
    yield Author(row)
  File "/usr/lib/python3.7/site-packages/scholarly/scholarly.py", line 226, in __init__
    self.name = __data.find('h3', class_='gsc_oai_name').text
AttributeError: 'NoneType' object has no attribute 'text'
```
OS: Manjaro Linux x86_64

**Comment by carlocaprini** [comment]
Should be easily resolved by renaming `gsc_oai_name ` to `gs_ai_name`.

**Comment by hack-r** [comment]
@carlocaprini The command was just `print(next(scholarly.search_author('Steven A. Cholewiak')))`. Do we need to edit the source code to try that change? 

If so, we should make a fork and submit a pull request...

**Comment by hack-r** [comment]
Actually, I don't think that did the trick...
```

    yield Author(row)
  File "C:\Users\hackr\Python64\Python372\lib\site-packages\scholarly\scholarly.py", line 226, in __init__
    self.name = __data.find('h3', class_='gsc_ai_name').text
AttributeError: 'NoneType' object has no attribute 'text'
```

**Comment by BananaLoaf** [comment]
@hack-r I made that mistake too, it's ```gs_ai_name```, not ```gsc_ai_name```

**Comment by hack-r** [comment]
@BannanaLoaf thank you! I haven't gotten home to try it yet but if this works that will be great.

**Comment by brodderickrodriguez** [comment]
Line 226 
original: self.name = __data.find('h3', class_='gsc_oai_name').tex
modification: self.name = __data.find('h3', class_='gs_ai_name').text

please update!

**Comment by datacustodian** [comment]
There are five instances that need to be updated: replaces all instances of "gsc_oai" with "gs_ai".

**Comment by Mustafallica** [comment]
getting same error

**Comment by bilbibaggins530** [comment]
I'm also getting the same error. Has anyone found a fix?

**Comment by jmlgh** [comment]
#40

**Comment by mbaddeley** [comment]
The gs_ai fix no longer works, not sure what the html has been changed to.

**Comment by jmlgh** [comment]
Just checked and it seems to be working fine. What did you try?

**Comment by mbaddeley** [comment]
oh... it now works... ok so that was weird. I must not have pressed save or something

**Comment by bilbibaggins530** [comment]
Just tried it and it works now! It wasn't working before with the gs_ai fix. It looks like Google changed their scholar site to be much more restrictive with the result that workarounds like Scholarly no longer seem to work reliably. Glad it's working now though.

**Comment by jmlgh** [comment]
For precisely this situations in the pull request I created last week, the class name is extracted dynamically, instead of being hard coded. If Google changes the class name again, Scholarly should continue working fine. 
#40

**Comment by maxsu** [comment]
Lets get this fix into pypi please.

I've tested jmlgh's fix:

> mkdir test; cd test
> echo "import scholarly \nprint(next(scholarly.search_author('Steven A. Cholewiak')))" > index.py
> git clone -b fixforissue37 https://github.com/jmlgh/scholarly.git
> pip install ./scholarly
> python index.py

Outputs:
```{'_filled': False,
 'affiliation': u'Assistant Project Scientist, University of California, Berkeley',
 'citedby': 206,
 'email': u'@berkeley.edu',
 'id': u'4bahYMkAAAAJ',
 'interests': [u'Depth Cues',
               u'3D Shape',
               u'Shape from Texture & Shading',
               u'Naive Physics',
               u'Haptics'],
 'name': u'Steven A. Cholewiak',
 'url_picture': u'https://scholar.google.com/citations?view_op=medium_photo&user=4bahYMkAAAAJ'}
```
>

**Comment by OrganicIrradiation** [comment]
Sorry for being absent.

I have been trying to test the merge but keep getting ReCAPTCHAed, so it's been slow to test. (Need a way to handle that ASAP...) Seems to all be good, but could you confirm it's working for you with jmlgh's fix merged:

```
pip install git+git://github.com/OrganicIrradiation/scholarly.git
```

**Comment by jmlgh** [comment]
After a clean install it seems to be working fine. 
By the way I would like to look into the ReCAPTCHAed thing. I've not run into it yet but probably will because I have to search for hundreds of publications in the project I'm currently working on. How can I test the ReCAPTCHAed site? What did you try?

**Comment by OrganicIrradiation** [comment]
Once your IP gets flagged, the only page thats returned will be the CAPTCHA. So, if you do enough inquiries (it would be helpful to remove the time.sleep that throttles requests), you'll find that the code raises exceptions since it no longer is pulling valid pages.

**Comment by bielsnohr** [comment]
Issue resolved by #40

**Comment by decause95** [comment]
Hi! 

Sorry but i'm stucked with this error. I've tried to do the "gs_ai" fix but I can't even find where to modify (if you haven't notice,  I'm still a new to programming). Any ideas please?

Thanks you

![image](https://user-images.githubusercontent.com/74783341/119399276-fee04300-bcd8-11eb-8f65-76d259c2bfe2.png)
